### PR TITLE
15 use aws secrets manager to store password for the rstudio user

### LINKED ISSUE #15: Use AWS Secrets Manager to store password for the rstudio-user

### PR DESCRIPTION
**Pull request by michaelmayer2**
This PR is addressing #15.

The new script in `deploy.sh` will ensure to create a new secret if it does not exist or will automatically update an existing secret (existing from a previous AWS ParallelCluster installation). 
Secret is labelled with cluster name which is a unique identifier for each ParallelCluster installation.
Upon cluster deployment, the `deploy.sh` will replace any occurrence of `SECRET` in the scripts `install-rsw.sh` and `install-compute.sh` with the new secret, copy the scripts to a S3 bucket and eventually AWS ParallelCluster will use those scripts to build the cluster. 

Sample output for `script.sh` that is the new part of `deploy.sh` below showing the creating and update capability of the new patch, 

```
michael@host % export CLUSTERNAME=test5
michael@host % bash script.sh
{
    "ARN": "arn:aws:secretsmanager:eu-west-1:637485797898:secret:rstudiopw-cluster-test5-UpRLDp",
    "Name": "rstudiopw-cluster-test5",
    "VersionId": "8d81a2cd-3b8d-4640-8ad2-0cb41bb4bb6c"
}
rstudio user password is now set to 3e935194

% bash script.sh

An error occurred (ResourceExistsException) when calling the CreateSecret operation: The operation failed because the secret rstudiopw-cluster-test5 already exists.
secret rstudiopw-cluster-test5 already exists, we need to update it
{
    "ARN": "arn:aws:secretsmanager:eu-west-1:637485797898:secret:rstudiopw-cluster-test5-UpRLDp",
    "Name": "rstudiopw-cluster-test5",
    "VersionId": "1592ea97-8321-4a70-a344-1bb97c5db6c5"
}
rstudio user password is now set to 2ef2264f
```

### CODE REVIEW

File: deploy.sh
Comment (jspiewak) [inline deploy.sh]: @aja08379 is this sufficient for your purposes?

File: scripts/install-compute.sh
Comment (jspiewak) [inline scripts/install-compute.sh:19]: Leave a comment saying that the value is replaced by the `deploy.sh` script at run time.

File: scripts/install-rsw.sh
Comment (jspiewak) [inline scripts/install-rsw.sh:143]: Ditto.

Comment (michaelmayer2) [reply scripts/install-rsw.sh:143]: Done. Does that look ok ?

File: scripts/install-compute.sh
Comment (michaelmayer2) [reply scripts/install-compute.sh:19]: Done. Does that look ok ?

File: deploy.sh
Comment (jspiewak) [inline deploy.sh:23]: I defer to @aja08379 as to whether this is sufficient.

Comment (aja08379) [reply deploy.sh:23]: @michaelmayer2 @jspiewak if this is coupled with the networking changes we mentioned (removing 0.0.0.0/0) then this looks fine to me. I would be surprised at this being brute-forced via the UI.

**Review by aja08379 — COMMENTED** [review]
The changes look OK to me. @jspiewak can you add Patrick to the org as I want to get his eyes on as well (username is @phalston). If these are coupled with the networking changes mentioned (removing 0.0.0.0/0) then I'd say this would be sufficient.

@michaelmayer2 is there any particular reason why the values generated cannot be longer? How are these used during testing?

**Comment by michaelmayer2** [comment]
> @michaelmayer2 is there any particular reason why the values generated cannot be longer? How are these used during testing?

@aja08379 I can make the values as long as you would like them to be - it may have overlapped your review but at the moment I am using a 12 character random string with a mix of numbers and upper+lower-case letters which should give us 12^(2*26+10) possibilities. The password value is being used in the RStudio Login prompt.

**Comment by aja08379** [comment]
@michaelmayer2 that sounds good to me. I have added Patrick to the org to review as well; once he's OK'd it I will approve.

**Comment by phalston** [comment]
LGTM.

**Review by aja08379 — APPROVED** [review]
Yup, looks good, approved.

**Pull request merged**
